table/table.go

```diff
@@ -101,108 +92,370 @@ func DefaultKeyMap() KeyMap {
+	return m
+}
+
+// SetHeight sets the width of the table. The given height will be the total
+// table height including borders, margins, and padding.
+func (m *Model) SetHeight(h int) *Model {
+	m.table.Height(h)
+	return m
+}
+
+// SetWidth sets the width of the table. The given width will be the total
+// table width including borders, margins, and padding.
+func (m *Model) SetWidth(w int) *Model {
+	m.table.Width(w)
+	return m
+}
+
+// SetFocused sets the focus state of the table.
+func (m *Model) SetFocused(f bool) *Model {
+	m.focus = f
+	return m
+}
+
+// SetYOffset sets the YOffset position in the table.
+func (m *Model) SetYOffset(n int) *Model {
+	m.yOffset = clamp(n, 0, len(m.rows)-1)
+	m.table.YOffset(m.yOffset)
+	return m
+}
+
+// SetStyles sets the table styles, only applying non-empty [Styles]. Note: using
+// [Model.SetStyleFunc] will override styles set in this function.
+func (m *Model) SetStyles(s Styles) *Model {
+	if !reflect.DeepEqual(s.Selected, lipgloss.Style{}) {
+		m.styles.Selected = s.Selected
+	}
+	if !reflect.DeepEqual(s.Header, lipgloss.Style{}) {
+		m.styles.Header = s.Header
+	}
+	if !reflect.DeepEqual(s.Cell, lipgloss.Style{}) {
+		m.styles.Cell = s.Cell
+	}
+	return m
+}
+
+// OverwriteStyles sets the table styles, overwriting all existing styles. Note:
+// using [Model.SetStyleFunc] will override styles set in this function.
+func (m *Model) OverwriteStyles(s Styles) *Model {
+	m.styles = s
+	return m
+}
+
+// OverwriteStylesFromLipgloss sets the [Model]'s style attributes from an
+// existing [lipgloss.Table].
+func (m *Model) OverwriteStylesFromLipgloss(t *table.Table) {
+	t.Rows(m.rows...)
+	t.Headers(m.headers...)
+	m.table = t
+	m.useStyleFunc = true
+}
+
+// SetStyleFunc sets the table's custom [table.StyleFunc]. Use this for conditional
+// styling e.g. styling a cell by its contents or by index.
+func (m *Model) SetStyleFunc(s table.StyleFunc) *Model {
+	m.useStyleFunc = true
+	m.table.StyleFunc(s)
+	return m
+}
+
+// Creation
+
+// New creates a new model for the table widget.
+func New(opts ...Option) *Model {
+	m := Model{
+		cursor: 0,
+		KeyMap: DefaultKeyMap(),
+		Help:   help.New(),
+		table:  table.New(),
+	}
+
+	m.SetStyles(DefaultStyles())
+
+	// Set border defaults here
+	m.Border(lipgloss.NormalBorder())
+	m.BorderTop(true)
+	m.BorderBottom(true)
+	m.BorderLeft(true)
+	m.BorderRight(true)
+	m.BorderColumn(false)
+	m.BorderRow(false)
+	m.BorderHeader(true)
+
+	for _, opt := range opts {
+		opt(&m)
+	}
+
+	return &m
+}
+
+// FromValues create the table rows from a simple string. It uses `\n` by
+// default for getting all the rows and the given separator for the fields on
+// each row.
+func (m *Model) FromValues(value, separator string) {
+	rows := [][]string{}
+	for _, line := range strings.Split(value, "\n") {
+		row := []string{}
+		for _, field := range strings.Split(line, separator) {
+			row = append(row, field)
+		}
+		rows = append(rows, row)
+	}
+
+	m.SetRows(rows...)
+}
```

Comment:
1. Do we need to add this?
2. If so, would it make sense to do a proper parse as CSV, to ensure char escaping work, etc?

Reply:
Removed this!